puppet/Dockerfile

```diff
@@ -0,0 +1,61 @@
+FROM phusion/baseimage:0.9.18
+
+MAINTAINER  eXo Platform <docker@exoplatform.com>
+
+# Set the locale
+RUN locale-gen en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+# eXo CI User
+ENV EXO_CI_USER ciagent
+ENV EXO_GROUP ciagent
+ENV HOME /home/${EXO_CI_USER}
+
+# Required directories for ciagent
+ENV EXO_CI_BASE       /opt/ciagent
+ENV EXO_CI_DATA_DIR   ${EXO_CI_BASE}/workspace
+ENV EXO_CI_LOG_DIR    ${EXO_CI_BASE}/logs
+ENV EXO_CI_TMP_DIR    ${EXO_CI_BASE}/tmp
+
+# Create user and group with specific ids
+RUN useradd --create-home --user-group -u 13000 --shell /bin/bash ${EXO_CI_USER}
+# giving all rights to eXo CI user
+RUN echo "ciagent  ALL = NOPASSWD: ALL" > /etc/sudoers.d/ciagent && chmod 440 /etc/sudoers.d/ciagent
+
+# Install puppet tools
+RUN apt-get update && apt-get install -y puppet && gem install r10k -v 2.4.3
```

Comment:
The puppet version installed is too old 3.4.x compared to the one we use.
Add these lines before puppet package install : 

```
# Install puppetlab repository
RUN gpg --keyserver keyserver.ubuntu.com --recv-key 4BD6EC30  \
  && gpg -a --export 4BD6EC30 | apt-key add -  \
  && echo "# Sources for recent versions of puppet" > /etc/apt/sources.list.d/puppetlab.list  \
  && echo "deb http://apt.puppetlabs.com $(lsb_release -c -s) main dependencies" >> /etc/apt/sources.list.d/puppetlab.list  \
  && echo "deb-src http://apt.puppetlabs.com $(lsb_release -c -s) main dependencies" >> /etc/apt/sources.list.d/puppetlab.list
```